cli/lib/enode.js

```diff
@@ -0,0 +1,67 @@
+const  Web3 = require("web3")
+
+exports.utils = {
+  toString: (enode) => {
+    enodeHigh = enode.enodeHigh.substring(2)
+    enodeLow = enode.enodeLow.substring(2)
+    ip = hexToIp(enode.ip)
+    port = enode.port.toNumber()
+    return `enode://${enodeHigh}${enodeLow}@${ip}:${port}`
+  },
+
+  parse: (enodeUrl) => {
+    url = new URL(enodeUrl)
+    pub = url.username
+
+    return {
+      originalUrl: url,
+      enodeHigh: '0x' + pub.substring(0, pub.length/2),
+      enodeLow: '0x' + pub.substring(pub.length/2),
+      hostname: getHexIpv4(url.hostname),
+      port: url.port
+    }
+  }
+}
+
+function getHexIpv4(stringIp) {
```

Comment:
All the code starting here is actually copypasted from https://github.com/AdharaProjects/permissioning-smart-contracts/blob/ignore_ip_port/src/util/ipConverter.ts